ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -667,11 +670,58 @@ public void shutdownTest() throws StoreException {
     verifyOperationFailuresOnInactiveStore(store);
     StorageManagerMetrics metrics = new StorageManagerMetrics(new MetricRegistry());
     StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
-    store = new BlobStore(storeId, config, scheduler, storeStatsScheduler, diskIOScheduler, metrics, tempDirStr,
-        LOG_CAPACITY, STORE_KEY_FACTORY, recovery, hardDelete, time);
+    store = new BlobStore(storeId, config, scheduler, storeStatsScheduler, diskIOScheduler, diskSpaceAllocator, metrics,
+        tempDirStr, LOG_CAPACITY, STORE_KEY_FACTORY, recovery, hardDelete, time);
     verifyOperationFailuresOnInactiveStore(store);
   }
 
+  /**
+   * Tests that {@link BlobStore#getDiskSpaceRequirements()} functions as expected.
+   * @throws StoreException
+   */
+  @Test
+  public void diskSpaceRequirementsTest() throws Exception {
+    // expect three log segments to be already allocated (from setup process)
+    int segmentsAllocated = 3;
+    doDiskSpaceRequirementsTest(segmentsAllocated, 0);
+
+    // try adding fake swap segment log segment.
+    File tempFile = File.createTempFile("sample-swap",
+        LogSegmentNameHelper.SUFFIX + BlobStoreCompactor.TEMP_LOG_SEGMENT_NAME_SUFFIX, tempDir);
+    doDiskSpaceRequirementsTest(segmentsAllocated, 1);
+    assertTrue("Could not delete temp file", tempFile.delete());
+
+    addCuratedData(SEGMENT_CAPACITY);
+    segmentsAllocated += 1;
+    doDiskSpaceRequirementsTest(segmentsAllocated, 0);
+
+    File.createTempFile("sample-swap",
+        LogSegmentNameHelper.SUFFIX + BlobStoreCompactor.TEMP_LOG_SEGMENT_NAME_SUFFIX, tempDir).deleteOnExit();
+    File.createTempFile("sample-swap",
+        LogSegmentNameHelper.SUFFIX + BlobStoreCompactor.TEMP_LOG_SEGMENT_NAME_SUFFIX, tempDir).deleteOnExit();
+    addCuratedData(SEGMENT_CAPACITY);
+    segmentsAllocated += 1;
+    doDiskSpaceRequirementsTest(segmentsAllocated, 2);
+  }
+
+  /**
+   * Run the {@link BlobStore#getDiskSpaceRequirements()} test.
+   * @param segmentsAllocated the number of segments currently used by the blob store.
+   * @param numSwapSegments the number of swap segments currently used by the blob store.
+   * @throws Exception
+   */
+  private void doDiskSpaceRequirementsTest(int segmentsAllocated, int numSwapSegments) throws Exception {
```

Comment:
minor: put this below the `//helpers` comment in the appropriate place

Reply:
moved